Formula/gptfdisk.rb

```diff
@@ -16,35 +16,43 @@ class Gptfdisk < Formula
 
   uses_from_macos "ncurses"
 
-  # update linker path for libncurses
-  patch :DATA
+  on_linux do
+    depends_on "util-linux"
+  end
 
   def install
-    system "make", "-f", "Makefile.mac"
+    on_macos do
+      inreplace "Makefile.mac" do |s|
+        s.gsub! "/usr/local/Cellar/ncurses/6.2/lib/libncurses.dylib", "-L/usr/lib -lncurses"
+        s.gsub! "-L/usr/local/lib -lpopt", "-L#{Formula["popt"].opt_lib} -lpopt"
+      end
+
+      system "make", "-f", "Makefile.mac"
+    end
+
+    on_linux do
+      deps = %w[ncurses popt util-linux]
+      ENV["CFLAGS"] = deps.map { |dep| "-I#{Formula[dep].opt_include}" }.join(" ")
+      ENV["CXXFLAGS"] = deps.map { |dep| "-I#{Formula[dep].opt_include}" }.join(" ")
```

Comment:
Although not entirely sure, does `ENV.append_to_cflags` work instead for adding to both?